course-mw/course-actors-common/src/main/java/org/sunbird/learner/util/ContentUtil.java

```diff
@@ -62,13 +62,24 @@ public static Map<String, Object> searchContent(String params, Map<String, Strin
             PropertiesCache.getInstance().getProperty(JsonKey.EKSTEP_AUTHORIZATION));
       }
       logger.info(null, "making call for content search ==" + params);
-      String response =
-          HttpUtil.sendPostRequest(
-              baseSearchUrl
-                  + PropertiesCache.getInstance().getProperty(JsonKey.EKSTEP_CONTENT_SEARCH_URL),
-              params,
-              headers);
-      logger.info(null, "Content search response", null, new HashMap<>(){{put("response", response);}});
+      String response = "";
+      if(Boolean.parseBoolean(ProjectUtil.getConfigValue(JsonKey.CONTENT_SERVICE_MOCK_ENABLED))){
```

Comment:
Rather than have the mock enabled check in ContentUtil, you should have some initialisation section during startup of the application which starts the mock server based on this flag. Content Util should not care about whether it is accessing actual or mock, it should just connect to the http endpoint and get a response. 

@BharathwajShankar

Reply:
noted @mathewjpallan will change accordingly

Reply:
@mathewjpallan the changes are addressed, can you please check now?